## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 440560`
Commit: `064ab48974d9792c0ebc6620cc48ee94a3ff8127`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>expliot</li>
    <li>expliot.dist</li>
    <li>python312Packages.dipy</li>
    <li>python312Packages.dipy.dist</li>
    <li>python313Packages.dipy</li>
    <li>python313Packages.dipy.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 207 packages built:</summary>
  <ul>
    <li>ants</li>
    <li>appcsxcad</li>
    <li>c3d</li>
    <li>cloudcompare</li>
    <li>csxcad</li>
    <li>elastix</li>
    <li>elmerfem</li>
    <li>exhibit</li>
    <li>ezminc</li>
    <li>f3d</li>
    <li>f3d.man</li>
    <li>freecad</li>
    <li>gdcm</li>
    <li>intensity-normalization (python313Packages.intensity-normalization)</li>
    <li>intensity-normalization.dist (python313Packages.intensity-normalization.dist)</li>
    <li>itk (itk_5)</li>
    <li>itk_5_2</li>
    <li>mirtk</li>
    <li>mrtrix</li>
    <li>octavePackages.dicom</li>
    <li>openems</li>
    <li>otb</li>
    <li>paraview</li>
    <li>pcl</li>
    <li>python312Packages.dcmstack</li>
    <li>python312Packages.dcmstack.dist</li>
    <li>python312Packages.deid</li>
    <li>python312Packages.deid.dist</li>
    <li>python312Packages.dicom-numpy</li>
    <li>python312Packages.dicom-numpy.dist</li>
    <li>python312Packages.dicom2nifti</li>
    <li>python312Packages.dicom2nifti.dist</li>
    <li>python312Packages.dicomweb-client</li>
    <li>python312Packages.dicomweb-client.dist</li>
    <li>python312Packages.f3d</li>
    <li>python312Packages.f3d.man</li>
    <li>python312Packages.fast-simplification</li>
    <li>python312Packages.fast-simplification.dist</li>
    <li>python312Packages.fslpy</li>
    <li>python312Packages.fslpy.dist</li>
    <li>python312Packages.gdcm</li>
    <li>python312Packages.gstools</li>
    <li>python312Packages.gstools.dist</li>
    <li>python312Packages.heudiconv</li>
    <li>python312Packages.heudiconv.dist</li>
    <li>python312Packages.highdicom</li>
    <li>python312Packages.highdicom.dist</li>
    <li>python312Packages.intensity-normalization</li>
    <li>python312Packages.intensity-normalization.dist</li>
    <li>python312Packages.itk</li>
    <li>python312Packages.mayavi</li>
    <li>python312Packages.mayavi.dist</li>
    <li>python312Packages.medpy</li>
    <li>python312Packages.medpy.dist</li>
    <li>python312Packages.medvol</li>
    <li>python312Packages.medvol.dist</li>
    <li>python312Packages.napari-nifti</li>
    <li>python312Packages.napari-nifti.dist</li>
    <li>python312Packages.nibabel</li>
    <li>python312Packages.nibabel.dist</li>
    <li>python312Packages.nilearn</li>
    <li>python312Packages.nilearn.dist</li>
    <li>python312Packages.nipy</li>
    <li>python312Packages.nipy.dist</li>
    <li>python312Packages.nipype</li>
    <li>python312Packages.nipype.dist</li>
    <li>python312Packages.nitime</li>
    <li>python312Packages.nitime.dist</li>
    <li>python312Packages.nitransforms</li>
    <li>python312Packages.nitransforms.dist</li>
    <li>python312Packages.niworkflows</li>
    <li>python312Packages.niworkflows.dist</li>
    <li>python312Packages.pybids</li>
    <li>python312Packages.pybids.dist</li>
    <li>python312Packages.pydicom</li>
    <li>python312Packages.pydicom.dist</li>
    <li>python312Packages.pykrige</li>
    <li>python312Packages.pykrige.dist</li>
    <li>python312Packages.pylibjpeg</li>
    <li>python312Packages.pylibjpeg-openjpeg</li>
    <li>python312Packages.pylibjpeg-openjpeg.dist</li>
    <li>python312Packages.pylibjpeg-rle</li>
    <li>python312Packages.pylibjpeg-rle.dist</li>
    <li>python312Packages.pylibjpeg.dist</li>
    <li>python312Packages.pymatgen</li>
    <li>python312Packages.pymatgen.dist</li>
    <li>python312Packages.pymedio</li>
    <li>python312Packages.pymedio.dist</li>
    <li>python312Packages.pynetdicom</li>
    <li>python312Packages.pynetdicom.dist</li>
    <li>python312Packages.pyorthanc</li>
    <li>python312Packages.pyorthanc.dist</li>
    <li>python312Packages.pyotb</li>
    <li>python312Packages.pyotb.dist</li>
    <li>python312Packages.pyradiomics</li>
    <li>python312Packages.pyradiomics.dist</li>
    <li>python312Packages.python-csxcad</li>
    <li>python312Packages.python-csxcad.dist</li>
    <li>python312Packages.python-openems</li>
    <li>python312Packages.python-openems.dist</li>
    <li>python312Packages.pyvista</li>
    <li>python312Packages.pyvista.dist</li>
    <li>python312Packages.sfepy</li>
    <li>python312Packages.sfepy.dist</li>
    <li>python312Packages.simpleitk</li>
    <li>python312Packages.simpleitk.dist</li>
    <li>python312Packages.sumo</li>
    <li>python312Packages.sumo.dist</li>
    <li>python312Packages.templateflow</li>
    <li>python312Packages.templateflow.dist</li>
    <li>python312Packages.torchio</li>
    <li>python312Packages.torchio.dist</li>
    <li>python312Packages.trx-python</li>
    <li>python312Packages.trx-python.dist</li>
    <li>python312Packages.vtk</li>
    <li>python312Packages.xnatpy</li>
    <li>python312Packages.xnatpy.dist</li>
    <li>python313Packages.dcmstack</li>
    <li>python313Packages.dcmstack.dist</li>
    <li>python313Packages.deid</li>
    <li>python313Packages.deid.dist</li>
    <li>python313Packages.dicom-numpy</li>
    <li>python313Packages.dicom-numpy.dist</li>
    <li>python313Packages.dicom2nifti</li>
    <li>python313Packages.dicom2nifti.dist</li>
    <li>python313Packages.dicomweb-client</li>
    <li>python313Packages.dicomweb-client.dist</li>
    <li>python313Packages.f3d</li>
    <li>python313Packages.f3d.man</li>
    <li>python313Packages.fast-simplification</li>
    <li>python313Packages.fast-simplification.dist</li>
    <li>python313Packages.fslpy</li>
    <li>python313Packages.fslpy.dist</li>
    <li>python313Packages.gdcm</li>
    <li>python313Packages.gstools</li>
    <li>python313Packages.gstools.dist</li>
    <li>python313Packages.heudiconv</li>
    <li>python313Packages.heudiconv.dist</li>
    <li>python313Packages.highdicom</li>
    <li>python313Packages.highdicom.dist</li>
    <li>python313Packages.itk</li>
    <li>python313Packages.medpy</li>
    <li>python313Packages.medpy.dist</li>
    <li>python313Packages.medvol</li>
    <li>python313Packages.medvol.dist</li>
    <li>python313Packages.napari-nifti</li>
    <li>python313Packages.napari-nifti.dist</li>
    <li>python313Packages.nibabel</li>
    <li>python313Packages.nibabel.dist</li>
    <li>python313Packages.nilearn</li>
    <li>python313Packages.nilearn.dist</li>
    <li>python313Packages.nipy</li>
    <li>python313Packages.nipy.dist</li>
    <li>python313Packages.nipype</li>
    <li>python313Packages.nipype.dist</li>
    <li>python313Packages.nitime</li>
    <li>python313Packages.nitime.dist</li>
    <li>python313Packages.nitransforms</li>
    <li>python313Packages.nitransforms.dist</li>
    <li>python313Packages.niworkflows</li>
    <li>python313Packages.niworkflows.dist</li>
    <li>python313Packages.pybids</li>
    <li>python313Packages.pybids.dist</li>
    <li>python313Packages.pydicom</li>
    <li>python313Packages.pydicom.dist</li>
    <li>python313Packages.pykrige</li>
    <li>python313Packages.pykrige.dist</li>
    <li>python313Packages.pylibjpeg</li>
    <li>python313Packages.pylibjpeg-openjpeg</li>
    <li>python313Packages.pylibjpeg-openjpeg.dist</li>
    <li>python313Packages.pylibjpeg-rle</li>
    <li>python313Packages.pylibjpeg-rle.dist</li>
    <li>python313Packages.pylibjpeg.dist</li>
    <li>python313Packages.pymedio</li>
    <li>python313Packages.pymedio.dist</li>
    <li>python313Packages.pynetdicom</li>
    <li>python313Packages.pynetdicom.dist</li>
    <li>python313Packages.pyorthanc</li>
    <li>python313Packages.pyorthanc.dist</li>
    <li>python313Packages.pyotb</li>
    <li>python313Packages.pyotb.dist</li>
    <li>python313Packages.pyradiomics</li>
    <li>python313Packages.pyradiomics.dist</li>
    <li>python313Packages.python-csxcad</li>
    <li>python313Packages.python-csxcad.dist</li>
    <li>python313Packages.pyvista</li>
    <li>python313Packages.pyvista.dist</li>
    <li>python313Packages.simpleitk</li>
    <li>python313Packages.simpleitk.dist</li>
    <li>python313Packages.templateflow</li>
    <li>python313Packages.templateflow.dist</li>
    <li>python313Packages.torchio</li>
    <li>python313Packages.torchio.dist</li>
    <li>python313Packages.trx-python</li>
    <li>python313Packages.trx-python.dist</li>
    <li>python313Packages.vtk</li>
    <li>python313Packages.xnatpy</li>
    <li>python313Packages.xnatpy.dist</li>
    <li>qcsxcad</li>
    <li>qcsxcad.dev</li>
    <li>rtabmap</li>
    <li>simpleitk</li>
    <li>therion</li>
    <li>vtk (vtk_9_5)</li>
    <li>vtk-dicom</li>
    <li>vtk-full</li>
    <li>vtkWithQt6</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.dipy</summary>
<pre>[34/123] Generating &#x27;dipy/tracking/streamlinespeed.cpython-312-x86_64-linux-gnu.so.p/streamlinespeed.c&#x27;
[35/123] Generating &#x27;dipy/denoise/denspeed.cpython-312-x86_64-linux-gnu.so.p/denspeed.c&#x27;
[36/123] Generating &#x27;dipy/direction/closest_peak_direction_getter.cpython-312-x86_64-linux-gnu.so.p/closest_peak_direction_getter.c&#x27;
[37/123] Generating &#x27;dipy/align/sumsqdiff.cpython-312-x86_64-linux-gnu.so.p/sumsqdiff.c&#x27;
[38/123] Generating &#x27;dipy/tracking/vox2track.cpython-312-x86_64-linux-gnu.so.p/vox2track.c&#x27;
[39/123] Generating &#x27;dipy/denoise/shift_twist_convolution.cpython-312-x86_64-linux-gnu.so.p/shift_twist_convolution.c&#x27;
[40/123] Generating &#x27;dipy/segment/mrf.cpython-312-x86_64-linux-gnu.so.p/mrf.c&#x27;
[41/123] Generating &#x27;dipy/tracking/localtrack.cpython-312-x86_64-linux-gnu.so.p/localtrack.c&#x27;
[42/123] Generating &#x27;dipy/core/interpolation.cpython-312-x86_64-linux-gnu.so.p/interpolation.c&#x27;
[43/123] Generating &#x27;dipy/segment/featurespeed.cpython-312-x86_64-linux-gnu.so.p/featurespeed.c&#x27;
[44/123] Generating &#x27;dipy/segment/metricspeed.cpython-312-x86_64-linux-gnu.so.p/metricspeed.c&#x27;
[45/123] Generating &#x27;dipy/align/crosscorr.cpython-312-x86_64-linux-gnu.so.p/crosscorr.c&#x27;
[46/123] Compiling C object dipy/utils/omp.cpython-312-x86_64-linux-gnu.so.p/meson-generated_omp.c.o
[47/123] Generating &#x27;dipy/align/vector_fields.cpython-312-x86_64-linux-gnu.so.p/vector_fields.c&#x27;
[48/123] Compiling C object dipy/align/bundlemin.cpython-312-x86_64-linux-gnu.so.p/meson-generated_bundlemin.c.o
[49/123] Compiling C object dipy/utils/tests/test_fast_numpy.cpython-312-x86_64-linux-gnu.so.p/meson-generated_test_fast_numpy.c.o
[50/123] Compiling C object dipy/align/transforms.cpython-312-x86_64-linux-gnu.so.p/meson-generated_transforms.c.o
ninja: build stopped: subcommand failed.

ERROR Backend subprocess exited when trying to invoke build_wheel</pre>
</details>
<details>
<summary>python313Packages.dipy</summary>
<pre>[30/123] Generating &#x27;dipy/utils/tests/test_fast_numpy.cpython-313-x86_64-linux-gnu.so.p/test_fast_numpy.c&#x27;
[31/123] Generating &#x27;dipy/reconst/recspeed.cpython-313-x86_64-linux-gnu.so.p/recspeed.c&#x27;
[32/123] Generating &#x27;dipy/reconst/eudx_direction_getter.cpython-313-x86_64-linux-gnu.so.p/eudx_direction_getter.c&#x27;
[33/123] Generating &#x27;dipy/segment/mrf.cpython-313-x86_64-linux-gnu.so.p/mrf.c&#x27;
[34/123] Generating &#x27;dipy/segment/cythonutils.cpython-313-x86_64-linux-gnu.so.p/cythonutils.c&#x27;
[35/123] Generating &#x27;dipy/direction/pmf.cpython-313-x86_64-linux-gnu.so.p/pmf.c&#x27;
[36/123] Generating &#x27;dipy/align/bundlemin.cpython-313-x86_64-linux-gnu.so.p/bundlemin.c&#x27;
[37/123] Generating &#x27;dipy/segment/clustering_algorithms.cpython-313-x86_64-linux-gnu.so.p/clustering_algorithms.c&#x27;
[38/123] Generating &#x27;dipy/denoise/nlmeans_block.cpython-313-x86_64-linux-gnu.so.p/nlmeans_block.c&#x27;
[39/123] Generating &#x27;dipy/tracking/propspeed.cpython-313-x86_64-linux-gnu.so.p/propspeed.c&#x27;
[40/123] Generating &#x27;dipy/tracking/direction_getter.cpython-313-x86_64-linux-gnu.so.p/direction_getter.c&#x27;
[41/123] Generating &#x27;dipy/direction/probabilistic_direction_getter.cpython-313-x86_64-linux-gnu.so.p/probabilistic_direction_getter.c&#x27;
[42/123] Generating &#x27;dipy/segment/metricspeed.cpython-313-x86_64-linux-gnu.so.p/metricspeed.c&#x27;
[43/123] Generating &#x27;dipy/core/interpolation.cpython-313-x86_64-linux-gnu.so.p/interpolation.c&#x27;
[44/123] Generating &#x27;dipy/align/sumsqdiff.cpython-313-x86_64-linux-gnu.so.p/sumsqdiff.c&#x27;
[45/123] Generating &#x27;dipy/align/crosscorr.cpython-313-x86_64-linux-gnu.so.p/crosscorr.c&#x27;
[46/123] Generating &#x27;dipy/align/vector_fields.cpython-313-x86_64-linux-gnu.so.p/vector_fields.c&#x27;
ninja: build stopped: subcommand failed.

ERROR Backend subprocess exited when trying to invoke build_wheel</pre>
</details>
</details>

--- a/pkgs/development/libraries/vtk/generic.nix
+++ b/pkgs/development/libraries/vtk/generic.nix
@@ -82,11 +82,10 @@
   libXcursor,
   gl2ps,
   libGL,
-  qt5,
   qt6,
 
   # custom options
-  qtVersion ? null,
+  withQt6 ? false,
   # To avoid conflicts between the propagated vtkPackages.hdf5
   # and the input hdf5 used by most downstream packages,
   # we set mpiSupport to false by default.
@@ -97,15 +96,6 @@
   testers,
 }:
 let
-  qtPackages =
-    if (isNull qtVersion) then
-      null
-    else if (qtVersion == "6") then
-      qt6
-    else if (qtVersion == "5") then
-      qt5
-    else
-      throw ''qtVersion must be "5", "6" or null'';
   vtkPackages = lib.makeScope newScope (self: {
     inherit
       tbb
@@ -174,7 +164,7 @@ stdenv.mkDerivation (finalAttrs: {
     libXrender
     libXcursor
   ]
-  ++ lib.optional (!(isNull qtPackages)) qtPackages.qttools
+  ++ lib.optional withQt6 qt6.qttools
   ++ lib.optional mpiSupport mpi
   ++ lib.optional pythonSupport tk;
 
@@ -235,12 +225,6 @@ stdenv.mkDerivation (finalAttrs: {
     })
   ];
 
-  # wrapper script calls qmlplugindump, crashes due to lack of minimal platform plugin
-  # Could not find the Qt platform plugin "minimal" in ""
-  preConfigure = lib.optionalString (qtVersion == "5") ''
-    export QT_PLUGIN_PATH=${lib.getBin qt5.qtbase}/${qt5.qtbase.qtPluginPrefix}
-  '';
-
   env = {
     CMAKE_PREFIX_PATH = "${lib.getDev openvdb}/lib/cmake/OpenVDB";
     NIX_LDFLAGS = "-L${lib.getLib libmysqlclient}/lib/mariadb";
@@ -276,8 +260,8 @@ stdenv.mkDerivation (finalAttrs: {
     (vtkBool "VTK_MODULE_ENABLE_VTK_RenderingOpenVR" false) # openvr
     (vtkBool "VTK_MODULE_ENABLE_VTK_RenderingAnari" false) # anari
 
-    # qtSupport
-    (vtkBool "VTK_GROUP_ENABLE_Qt" (!(isNull qtPackages)))
+    # withQt6
+    (vtkBool "VTK_GROUP_ENABLE_Qt" withQt6)
     (lib.cmakeFeature "VTK_QT_VERSION" "Auto") # will search for Qt6 first
 
     # pythonSupport
@@ -325,9 +309,9 @@ stdenv.mkDerivation (finalAttrs: {
 
         package = finalAttrs.finalPackage;
 
-        nativeBuildInputs = lib.optionals (!(isNull qtPackages)) [
-          qtPackages.qttools
-          qtPackages.wrapQtAppsHook
+        nativeBuildInputs = lib.optionals withQt6 [
+          qt6.qttools
+          qt6.wrapQtAppsHook
         ];
       };
     };

--- a/pkgs/top-level/aliases.nix
+++ b/pkgs/top-level/aliases.nix
@@ -2562,7 +2562,8 @@ mapAliases {
   voxelands = throw "'voxelands' has been removed due to lack of upstream maintenance"; # Added 2025-08-30
   vtk_9 = lib.warnOnInstantiate "'vtk_9' has been renamed to 'vtk_9_5'" vtk_9_5; # Added 2025-07-18
   vtk_9_egl = lib.warnOnInstantiate "'vtk_9_5' now build with egl support by default, so `vtk_9_egl` is deprecated, consider using 'vtk_9_5' instead." vtk_9_5; # Added 2025-07-18
-  vtk_9_withQt5 = throw "'vtk_9_withQt5' has been removed, Consider using 'vtkWithQt5' instead."; # Added 2025-07-18
+  vtk_9_withQt5 = throw "'vtk_9_withQt5' has been removed, Consider using 'vtkWithQt6' instead."; # Added 2025-07-18
+  vtkWithQt5 = throw "'vtkWithQt5' has been removed. Consider using 'vtkWithQt6' instead."; # Added 2025-09-06
   vuze = throw "'vuze' was removed because it is unmaintained upstream and insecure (CVE-2018-13417). BiglyBT is a maintained fork."; # Added 2024-11-22
   vwm = throw "'vwm' was removed as it is broken and not maintained upstream"; # Added 2025-05-17
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -9046,14 +9046,12 @@ with pkgs;
   vtk = vtk_9_5;
 
   vtk-full = vtk.override {
-    qtVersion = "6";
+    withQt6 = true;
     mpiSupport = true;
     pythonSupport = true;
   };
 
-  vtkWithQt5 = vtk.override { qtVersion = "5"; };
-
-  vtkWithQt6 = vtk.override { qtVersion = "6"; };
+  vtkWithQt6 = vtk.override { withQt6 = true; };
 
   vulkan-caps-viewer = libsForQt5.callPackage ../tools/graphics/vulkan-caps-viewer { };
 

